src/renderers/shaders/ShaderChunk/lights_phong_fragment.glsl

```diff
@@ -78,32 +51,37 @@ vec3 totalSpecularLight = vec3( 0.0 );
 
 	for( int i = 0; i < MAX_DIR_LIGHTS; i ++ ) {
 
-		vec3 lightColor = directionalLightColor[ i ];
+		IncidentLight incidentLight;
+		getDirLightDirect( directionalLights[ i ], incidentLight );
 
-		vec3 lightDir = directionalLightDirection[ i ];
+		totalDirectReflectedDiffuse +=
+			BRDF_Lambert( incidentLight, normal, diffuse );
 
-		// diffuse
+		totalDirectReflectedSpecular +=
+			BRDF_BlinnPhong( incidentLight, normal, viewDir, specular, shininess );
 
-		float cosineTerm = saturate( dot( normal, lightDir ) );
+	}
 
-		totalDiffuseLight += lightColor * cosineTerm;
+#endif
+
+#if MAX_HEMI_LIGHTS > 0
 
-		// specular
+	for( int i = 0; i < MAX_HEMI_LIGHTS; i ++ ) {
 
-		vec3 brdf = BRDF_BlinnPhong( specular, shininess, normal, lightDir, viewDir );
+		IncidentLight incidentLight;
+		getHemisphereLightIndirect( hemisphereLights[ i ], normal, incidentLight );
 
-		totalSpecularLight += brdf * specularStrength * lightColor * cosineTerm;
+		totalIndirectReflectedDiffuse +=
+			BRDF_Lambert( incidentLight, normal, diffuse );
 
 	}
 
 #endif
 
-#ifdef METAL
-
-	outgoingLight += diffuseColor.rgb * ( totalDiffuseLight + totalAmbientLight ) * specular + totalSpecularLight + totalEmissiveLight;
 
-#else
-
-	outgoingLight += diffuseColor.rgb * ( totalDiffuseLight + totalAmbientLight ) + totalSpecularLight + totalEmissiveLight;
-
-#endif
+outgoingLight +=
+	totalDirectReflectedDiffuse +
+	totalDirectReflectedSpecular +
+	totalIndirectReflectedDiffuse +
+	totalIndirectReflectedSpecular +
+	totalEmissiveLight;
```

Comment:
Yes, it is critical that they are accumulated separately this way.  Remember you are looking at a work in progress.  I still need to extend the usage of this division in the accumulation outside of just this *.glsl file, but I am doing it incrementally so I have working code at all the intermediate steps.

When one wants to incorporate SAO, SSAO, aoMaps, one needs to dim only the indirect components, and sometimes only indirect diffuse.  In a later commit, I have introduced a ReflectedLight struct to make this a bit easier to deal with.